crates/re_space_view_spatial/src/parts/images.rs

```diff
@@ -681,13 +681,19 @@ impl ViewPartSystem for ImagesPart {
         &self,
         store: &re_arrow_store::DataStore,
         ent_path: &EntityPath,
+        ctx: HeuristicFilterContext,
         query: &LatestAtQuery,
         entity_components: &ComponentNameSet,
     ) -> bool {
         if !default_heuristic_filter(entity_components, &self.indicator_components()) {
             return false;
         }
 
+        // If this is a 3D view and there's no parent pinhole, do not include this part.
+        if ctx.class == "3D" && !ctx.has_ancestor_pinhole {
+            return false;
+        }
```

Comment:
Why though?

Reply:
I can already read the Rust code and see _what_ it does, I want to know _why_ it does it!

Reply:
https://github.com/rerun-io/rerun/pull/3836